src/features/vars-schema/env-vars.ts

```diff
@@ -56,7 +56,7 @@ export const BLOCKED_NAMES = addReservedPrefixes([
 	'HOST_LOG_TO_DISPLAY',
 ]);
 
-export const INVALID_CHARACTER_REGEX = /^\d|\W/;
+export const INVALID_CHARACTER_REGEX = /^(\d|:)|[^A-Za-z0-9_:]/;
```

Comment:
It's best to avoid the capturing group as it's unnecessary, either with `[...]`
```suggestion
export const INVALID_CHARACTER_REGEX = /^[\d:]|[^A-Za-z0-9_:]/;
```
or mark it as a non-capturing group `(?:...)`
```suggestion
export const INVALID_CHARACTER_REGEX = /^(?:\d|:)|[^A-Za-z0-9_:]/;
```